common/common/media.py

```diff
@@ -245,3 +273,110 @@ def _request(url, max_size, content_types):
 			raise TooLarge(f"Content length {length} is too large for url {url}")
 
 	return resp
+
+
+def download_imgur_url(output_dir, max_size, url):
+	"""Links to imgur require special handling to resolve the actual image.
+	Handles URLs like the following:
+		i.stack.imgur.com/ID.png
+		imgur.com/ID
+		i.imgur.com/ID.EXT
+			These map to actual media and are stored in the usual way.
+		imgur.com/a/ID
+		imgur.com/gallery/ID
+			These map to collections of media.
+			Under the original URL we store a json file that lists imgur.com/ID urls
+			of the contents of the collection. Those urls are then downloaded and stored
+			in the usual way.
+	"""
+	parsed = urllib.parse.urlparse(url)
+	if parsed.hostname not in ("imgur.com", "i.stack.imgur.com"):
```

Comment:
What about `i.imgur.com`? (Mentioned in the docstring but not here)

Reply:
`i.imgur.com` requires no special handling (it's a direct link to the image) so we don't need to handle it here.

Reply:
I can clarify that in the docstring, you're right it does say this function "handles" that case.